app/controllers/reauthn_required_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ReauthnRequiredController < ApplicationController
+  before_action :confirm_recently_authenticated
+
+  def recently_authenticated?
+    return false unless user_session.present?
+    authn_at = user_session[:authn_at]
```

Comment:
throw this into a private method so `recently_authenticated?` is a tad shorter / easier to read?